src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private static final Random random = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        final int figureType = random.nextInt(FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+        int sideOrRadius = random.nextInt() * 10;
+        switch (figureType) {
+            case 0:
+                return new Square(color, sideOrRadius);
+            case 1:
+                return new Circle(color, sideOrRadius);
+            case 2:
+                double width = random.nextDouble() * 10;
+                double height = random.nextDouble() * 10;
+                return new Rectangle(color, (int) width, (int) height);
+            case 3:
+                double firstLeg = random.nextDouble() * 10;
+                double secondLeg = random.nextDouble() * 10;
+                return new RightTriangle(color, (int) secondLeg, (int) firstLeg);
+            case 4:
+                int topBase = random.nextInt() * 10;
+                int bottomBase = random.nextInt() * 10;
+                int heightTrapezoid = random.nextInt() * 10;
+                return new IsoscelesTrapezoid(color, topBase, bottomBase, heightTrapezoid);
```

Comment:
For every case you can create own method and you have a lot of duplicate code with your random values